tests/sgl/func/test_tensor.cpp

```
@@ -0,0 +1,146 @@
+    BufferDesc buffer_desc;
+    buffer_desc.usage = usage;
+    buffer_desc.memory_type = memory_type;
+    buffer_desc.struct_size = type.element_layout->stride();
+    buffer_desc.element_count = element_count;
+    return device->create_buffer(buffer_desc);
+}
+
+ref<func::Tensor> make_tensor(
+    Device* device,
+    const TensorTestType& type,
+    slangpy::Shape shape,
```

Cppcheck (2.20.0)

[performance] Function parameter 'shape' should be passed by const reference.

(passedByValue)